TurnkeyReact/Oauth.tsx

```diff
@@ -0,0 +1,143 @@
+import { useTurnkey } from "@turnkey/sdk-react-native";
+import { useEffect, useState, useCallback } from "react";
+import { SharedConfigs } from "./SharedConfigs";
+import { Button } from "./ui/button";
+import { View, Text } from "react-native";
+// import GoogleIcon from "../../assets/svgs/google.svg";
+import { OAUTH_TOKEN_EXPIRATION_SECONDS } from "./lib/constants";
+// import * as Crypto from "expo-crypto";
+import { styles } from "./TurnkeyStyle";
+import SHA256 from 'crypto-js/sha256';
+
+interface OAuthProps {
+  onSuccess: (params: {
+    oidcToken: string;
+    providerName: string;
+    targetPublicKey: string;
+    expirationSeconds: string;
+  }) => Promise<void>;
+  configs: SharedConfigs
+}
+
+interface AuthButtonProps extends OAuthProps {
+  nonce: string | null;
+  targetPublicKey: string | null;
+  refreshNonce: () => Promise<void>;
+}
```

Comment:
nit: prefer using `type` over `interface`

Then instead of extends you can use `&`

```suggestion
type OAuthProps =  {
  onSuccess: (params: {
    oidcToken: string;
    providerName: string;
    targetPublicKey: string;
    expirationSeconds: string;
  }) => Promise<void>;
  configs: SharedConfigs;
}

type AuthButtonProps = {
  nonce: string | null;
  targetPublicKey: string | null;
  refreshNonce: () => Promise<void>;
} & OAuthProps;
```